task-impls/src/da.rs

```diff
@@ -0,0 +1,334 @@
+                    Ok(Some(vote_token)) => {
+                        info!("We were chosen for DA committee on {:?}", view);
+
+                        // Generate and send vote
+                        let message = self.committee_exchange.create_da_message::<I>(
+                            self.high_qc.commit(),
+                            block_commitment,
+                            view,
+                            vote_token,
+                        );
+
+                        self.cur_view = view;
+
+                        if let CommitteeConsensusMessage::DAVote(vote) = message {
+                            info!("Sending vote to the DA leader {:?}", vote);
+                            self.event_stream
+                                .publish(SequencingHotShotEvent::DAVoteSend(vote))
+                                .await;
+                        }
+                    }
+                }
+            }
+            SequencingHotShotEvent::DAVoteRecv(vote, sender) => {
+                if vote.signature.0 != <TYPES::SignatureKey as SignatureKey>::to_bytes(&sender) {
+                    return None;
+                }
+                let handle_event = HandleEvent(Arc::new(move |event, state| {
+                    async move { vote_handle(state, event).await }.boxed()
+                }));
+                let (collection_view, _collection_task) = &self.vote_collector;
+                let acc = VoteAccumulator {
+                    total_vote_outcomes: HashMap::new(),
+                    yes_vote_outcomes: HashMap::new(),
+                    no_vote_outcomes: HashMap::new(),
+                    success_threshold: self.committee_exchange.success_threshold(),
+                    failure_threshold: self.committee_exchange.failure_threshold(),
+                    viewsync_precommit_vote_outcomes: HashMap::new(),
+                };
+                // Todo check if we are the leader
+                let accumulator = self.committee_exchange.accumulate_vote(
+                    &vote.signature.0,
+                    &vote.signature.1,
+                    vote.block_commitment,
+                    vote.vote_data,
+                    vote.vote_token.clone(),
+                    vote.current_view,
+                    acc,
+                    None,
+                );
+                if vote.current_view > *collection_view {
+                    let state = DAVoteCollectionTaskState {
+                        committee_exchange: self.committee_exchange.clone(),
+                        accumulator,
+                        cur_view: vote.current_view,
+                        event_stream: self.event_stream.clone(),
+                    };
+                    let name = "DA Vote Collection";
+                    let filter = FilterEvent::default();
+                    // TODO (run_view refactor) `TaskBuilder` is created but the task isn't added to the task runner.
+                    let _builder =
```

Comment:
I updated the consensus task vote collection task logic to include the registering and running it in my pr if you want to have a look.  We should also add in our tech debt clean up to reuse the code for vote collection.  Looks like this is done in all 3 of the main tasks (excluding the network one)